core/state/statedb.go

```diff
@@ -679,3 +681,99 @@ func (db *DB) Commit(deleteEmptyObjects bool) (root common.Hash, err error) {
 	//log.Debug("Trie cache stats after commit", "misses", trie.CacheMisses(), "unloads", trie.CacheUnloads())
 	return root, err
 }
+
+// GetValidatorMap returns the staking validators, if not exist, create one
+func (db *DB) GetValidatorMap() map[common.Address]struct{} {
+	by := db.GetCode(staking.ValidatorMapAddress)
```

Comment:
This list should be piggybacked on a state object. It should be stored in local db (offchain). Because it's derived data from the all the validator object. Duplicating it in the state is not good.